src/rankcorr.jl

```diff
@@ -17,14 +17,108 @@ Compute Spearman's rank correlation coefficient. If `x` and `y` are vectors, the
 output is a float, otherwise it's a matrix corresponding to the pairwise correlations
 of the columns of `x` and `y`.
 """
-corspearman(x::RealVector, y::RealVector) = cor(tiedrank(x), tiedrank(y))
+function corspearman(x::RealVector, y::RealVector)
+    n = length(x)
+    n == length(y) || throw(DimensionMismatch("vectors must have same length"))
+    (any(isnan, x) || any(isnan, y)) && return NaN
+    return cor(tiedrank(x), tiedrank(y))
+end
+
+function corspearman(X::RealMatrix, y::RealVector)
+    size(X, 1) == length(y) ||
+        throw(DimensionMismatch("X and y have inconsistent dimensions"))
+    n = size(X, 2)
+    C = Matrix{Float64}(I, n, 1)
+    any(isnan, y) && return fill!(C, NaN)
+    yrank = tiedrank(y)
+    for j = 1:n
+        Xj = view(X, :, j)
+        if any(isnan, Xj)
+            C[j,1] = NaN
+        else
+            Xjrank = tiedrank(Xj)
+            C[j,1] = cor(Xjrank, yrank)
+        end
+    end
+    return C
+end
 
-corspearman(X::RealMatrix, Y::RealMatrix) =
-    cor(mapslices(tiedrank, X, dims=1), mapslices(tiedrank, Y, dims=1))
-corspearman(X::RealMatrix, y::RealVector) = cor(mapslices(tiedrank, X, dims=1), tiedrank(y))
-corspearman(x::RealVector, Y::RealMatrix) = cor(tiedrank(x), mapslices(tiedrank, Y, dims=1))
+function corspearman(x::RealVector, Y::RealMatrix)
+    size(Y, 1) == length(x) ||
+        throw(DimensionMismatch("x and Y have inconsistent dimensions"))
+    n = size(Y, 2)
+    C = Matrix{Float64}(I, 1, n)
+    any(isnan, x) && return fill!(C, NaN)
+    xrank = tiedrank(x)
+    for j = 1:n
+        Yj = view(Y, :, j)
+        if any(isnan, Yj)
+            C[1,j] = NaN
+        else
+            Yjrank = tiedrank(Yj)
+            C[1,j] = cor(xrank, Yjrank)
+        end
+    end
+    return C
+end
+
+function corspearman(X::RealMatrix)
+    n = size(X, 2)
+    C = Matrix{Float64}(I, n, n)
+    anynan = Vector{Union{Bool, Missing}}(missing, n)
+    for j = 1:n
+        Xj = view(X, :, j)
+        if ismissing(anynan[j])
+            anynan[j] = any(isnan, Xj)
+        end
+        if anynan[j]
+            C[:,j] .= NaN
+            C[j,:] .= NaN
+            C[j,j] = 1
+            continue
+        end
+        Xjrank = tiedrank(Xj)
+        for i = 1:(j-1)
+            Xi = view(X, :, i)
+            if ismissing(anynan[i])
```

Comment:
this line is not covered by tests as in this point `anynan[i]` may not be `missing` as `i in 1:(j-1)`. Right?

Reply:
Indeed. For some reason I though the pattern was more complex. :-)